src/mainwindow.cpp

```diff
@@ -148,6 +148,21 @@ MainWindow::MainWindow(QWidget* parent)
     connect(m_startPage, &StartPage::recordButtonClicked, this, &MainWindow::onRecordButtonClicked);
     connect(m_startPage, &StartPage::stopParseButtonClicked, this,
             static_cast<void (MainWindow::*)()>(&MainWindow::clear));
+    connect(m_startPage, &StartPage::diffButtonClicked, this, [this] {
+        const auto fileNameA = QFileDialog::getOpenFileName(this, tr("Open File A"), QDir::currentPath(),
+                                                            tr("Data Files (perf*.data perf.data.*);;All Files (*)"));
```

Comment:
share code with below

and ideally introduce a custom QDialog that uses two KUrlRequester to specify the files, instead of showing two modal dialogs

Reply:
still open

Reply:
there is now a custom dialog